src/util/string.h

```diff
@@ -31,9 +31,10 @@ NODISCARD inline std::string TrimString(const std::string& str, const std::strin
  * @param unary_op   Apply this operator to each item in the list
  */
 template <typename T, typename UnaryOp>
-std::string Join(const std::vector<T>& list, const std::string& separator, UnaryOp unary_op)
+auto Join(const std::vector<T>& list, const std::string& separator, UnaryOp unary_op)
```

Comment:
```suggestion
auto Join(const std::vector<T>& list, const U& separator, UnaryOp unary_op)
```

I think this requirement can be lifted (here and below)

Reply:
[Updated](https://github.com/bitcoin/bitcoin/pull/18922#issuecomment-626197278).